### PR TITLE
macos: Delete build_prefix rpath

### PR DESCRIPTION
**Pull request by jschueller**
Leaving invalid build_prefix rpath entries is fine in general but
may confuse tools such as delocate.

cc @isuruf

### CODE REVIEW

**Review by isuruf — CHANGES_REQUESTED** [review]
This should be done by getting all the rpaths and then checking if any of them starts with `build_prefix` and then deleting them. there is `macho.get_rpaths` to get all the rpaths

**Comment by jschueller** [comment]
ok, I guess install_name_tool can throw an error if its not there

**Comment by isuruf** [comment]
Can you use my branch `rpath`  in github.com/isuruf/conda-build ?

**Comment by jschueller** [comment]
ok, I cherry-picked it from your branch and squashed it

**Comment by isuruf** [comment]
Can you please keep me as the git author?

**Comment by jschueller** [comment]
I think it was lost due to the squashing the 2 commits

**Comment by isuruf** [comment]
Thanks @jschueller for fixing it

**Comment by jschueller** [comment]
the ci is failling, but it doesnt look related

File: conda_build/os_utils/macho.py [hunk 1]
```diff
@@ -218,10 +218,10 @@ def get_id(path, build_prefix=None):
         return ''
 
 
-def get_rpaths(path):
+def get_rpaths(path, build_prefix=None):
     """Return a list of the dylib rpaths"""
-    res_pyldd = inspect_rpaths(path, resolve_dirnames=False, use_os_varnames=True)
-    return res_pyldd
+    dylib_loads = otool(path, build_prefix, is_rpath)
+    return [dylib_load['path'] for dylib_load in dylib_loads]
```
Comment (mingwandroid) [inline conda_build/os_utils/macho.py:225]: why this change?

Comment (isuruf) [reply conda_build/os_utils/macho.py:225]: `delete_rpath` and `insert_rpath` were both using `otool` instead of `pyldd`. Should this be reverted?

Comment (mingwandroid) [reply conda_build/os_utils/macho.py:225]: Not sure, I'll check a bit later. In general I'm trying to move things towards our own implementations but let me get back to you on this. Thanks for all the effort you are all putting in to conda-build by the way. We at Anaconda greatly appreciate it.

Comment (jschueller) [reply conda_build/os_utils/macho.py:225]: @mingwandroid any news on this ?

Comment (mingwandroid) [reply conda_build/os_utils/macho.py:225]: Sorry not yet. I'll make sure to review it before the next release though. Thanks

Comment (mingwandroid) [reply conda_build/os_utils/macho.py:225]: OK, I'm looking into the release now. It's grown in scope a lot, sorry about the delay @jschueller and @isuruf

**Comment by jschueller** [comment]
was cherry-picked
